Pytest v7.1 fails to initialize on windows

While the CI is working fine, the test suite is failing to run on windows. The current quick-fix is to downgrade to pytest v7.0, but this issue should be fixed.

Example output of error:
``` python
================================================= test session starts =================================================
platform win32 -- Python 3.10.4, pytest-7.1.2, pluggy-1.0.0
django: settings: simmate.configuration.django.settings_test (from ini)
rootdir: C:\Users\jacks\Documents\GitHub\simmate, configfile: pyproject.toml
plugins: django-4.5.2, forked-1.4.0, mock-3.7.0, xdist-2.5.0
collected 0 items / 1 error

======================================================= ERRORS ========================================================
____________________________________________ ERROR collecting test session ____________________________________________
..\..\..\anaconda3\envs\simmate_dev\lib\site-packages\_pytest\runner.py:338: in from_call
    result: Optional[TResult] = func()
..\..\..\anaconda3\envs\simmate_dev\lib\site-packages\_pytest\runner.py:369: in <lambda>
    call = CallInfo.from_call(lambda: list(collector.collect()), "collect")
..\..\..\anaconda3\envs\simmate_dev\lib\site-packages\_pytest\main.py:723: in collect
    for x in self._collectfile(pkginit):
..\..\..\anaconda3\envs\simmate_dev\lib\site-packages\_pytest\main.py:575: in _collectfile
    ihook = self.gethookproxy(fspath)
..\..\..\anaconda3\envs\simmate_dev\lib\site-packages\_pytest\main.py:539: in gethookproxy
    my_conftestmodules = pm._getconftestmodules(
..\..\..\anaconda3\envs\simmate_dev\lib\site-packages\_pytest\config\__init__.py:575: in _getconftestmodules
    mod = self._importconftest(conftestpath, importmode, rootpath)
..\..\..\anaconda3\envs\simmate_dev\lib\site-packages\_pytest\config\__init__.py:620: in _importconftest
    assert mod not in mods
E   AssertionError
================================================== warnings summary ===================================================
..\..\..\anaconda3\envs\simmate_dev\lib\site-packages\past\builtins\misc.py:45
  C:\Users\jacks\anaconda3\envs\simmate_dev\lib\site-packages\past\builtins\misc.py:45: DeprecationWarning: the imp module is deprecated in favour of importlib and slated for removal in Python 3.12; see the module's documentation for alternative uses
    from imp import reload

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
=============================================== short test summary info ===============================================
ERROR  - AssertionError
!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!! Interrupted: 1 error during collection !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
============================================= 1 warning, 1 error in 0.91s =============================================
```

--- a/pyproject.toml
+++ b/pyproject.toml
@@ -94,7 +94,7 @@ dependencies=[
 
 # For development and testing
 DEV = [
-    "pytest >=6.2.5, <7.1",  # BUG: see issue #162 for limiting <7.1
+    "pytest >=6.2.5, <=7.1.2",
     "pytest-django >=4.5.2, <=4.5.2",
     "pytest-mock >=3.7.0, <=3.8.2",
     "pytest-xdist >=2.5.0, <=2.5.0",

--- a/tutorials/Guides_for_contributors/environment.yaml
+++ b/tutorials/Guides_for_contributors/environment.yaml
@@ -31,7 +31,7 @@ dependencies:
     - plotly >=5.4.0, <=5.9.0
     - pymatgen >=2022.1.9, <=2022.7.25
     - pymatgen-analysis-diffusion >=2021.4.29, <=2022.7.21
-    - pytest >=6.2.5, <7.1
+    - pytest >=6.2.5, <=7.1.2
     - pytest-django >=4.5.2, <=4.5.2
     - python >=3.10
     - pytest-mock >=3.7.0, <=3.8.2

